server/src/main/java/org/apache/druid/client/BrokerSegmentWatcherConfig.java

```diff
@@ -46,6 +49,11 @@
     return watchedDataSources;
   }
 
+  public boolean isWatchRealtimeNodes()
+  {
+    return watchRealtimeNodes;
+  }
```

Comment:
I think my only real question about this PR is if this should maybe be `watchedServerTypes` or something and take a list of server types to watch instead of just a boolean

Reply:
I was thinking the same thing but then I realized that only a few combinations are possible here:
(a) REALTIME and HISTORICAL, (b) only HISTORICAL (and maybe (c)  only REALTIME).

Could there be other queryable types in the future (or even now that I may have missed)?. If not, then I guess with the boolean, we just miss out on option (c) above.

Please let me know what you think.

Reply:
if you have broadcast load rules, and segment cache configured, brokers can load segments as well. They currently have independent handling in `BrokerServerView` (https://github.com/apache/druid/blob/master/server/src/main/java/org/apache/druid/client/BrokerServerView.java#L226), but i think could in theory also be filtered here.

If we did do a list, the default value should probably be like, all the server types.

Reply:
I'm not necessarily advocating for it to be a list.. just thinking out loud. A deny list might be better if we do go that route.

Reply:
`watchedServerTypes` seems too broad and technical. Trying to understand the use case behind such a config. e.g. what would a `watchedServerTypes=Broker` would mean?

Reply:
yeah, I guess it has the problem of being able to craft incorrect configurations, which is lame. 

Side note, would just setting the realtime tasks to use a different tier than historicals and just using `watchedTiers` not flexible enough to solve the issue of wanting to ignore realtime tasks? I guess the drawback would be if you _were_ using broadcast load rules then you'd need to load on both historical and realtime tiers, but that doesn't seem terrible.

Reply:
hmm. that change will increase the scope of this PR. then users aren't used to thinking of real-time nodes as a tier.

Reply:
>hmm. that change will increase the scope of this PR. then users aren't used to thinking of real-time nodes as a tier.

Ah, I meant using existing functionality prior to this PR, wasn't sure if it would work or not

Reply:
so we would set something like `druid.server.tier=_realtime` on MM/indexers and then not include that in the watchedTiers? Is that what you are suggesting or something else?

Reply:
yes, that exactly is what i was wondering if would work

Reply:
it is worth trying. Hope it doesn't break something else 😄